src/OffsetArrays.jl

```diff
@@ -347,15 +347,24 @@ end
 
 # Reshaping OffsetArrays can "pop" the original OffsetArray wrapper and return
 # an OffsetArray(reshape(...)) instead of an OffsetArray(reshape(OffsetArray(...)))
+# try to pass on the indices as received to the parent.
+# If the parent doesn't define an appropriate method, this will fall back to using the lengths
+# Short-circuit the case with matching indices to circumvent the Base restriction to 1-based indices
+_reshape(A::AbstractArray{<:Any,N}, ::NTuple{N,Union{Integer, AbstractUnitRange}}) where {N} = A
+_reshape(A::AbstractArray{<:Any,N}, inds::NTuple{N,OffsetAxis}) where {N} = (_colon_check(inds); A)
+_reshape(A, inds) = reshape(A, inds)
+_reshape_nov(A, inds) = no_offset_view(_reshape(A, inds))
```

Comment:
It is subtle and can be fragile to require the caller to call `_reshape_nov(parent(A), inds)`.

I believe this one is equivalent?

```suggestion
_reshape_nov(A, inds) = _reshape(no_offset_view(A), inds)
```

Reply:
This seems to be equivalent, so I've changed it to this now